spec/lib/kiev/shoryuken_spec.rb

```diff
@@ -0,0 +1,174 @@
+
+        it "tags it with tree_path B (first request, asynchronous)" do
+          expect(sqs)
+            .to have_received(:send_message)
+            .with(hash_including(message_attributes: { "tree_path" => { data_type: "String", string_value: "B" } }))
+        end
+
+        it "assigns some request_id to it" do
+          skip("Is this necessary? If so, it has to happen everywhere")
+        end
+      end
+
+      context "with a specified request_id in Kiev store" do
+        let(:request_id) { "acc6acfe-525e-49a6-a12b-ce3f07564620" }
+
+        before do
+          Kiev::RequestStore.store[:request_id] = request_id
+          send_message
+        end
+
+        it "preserves request_id" do
+          expect(sqs)
+            .to have_received(:send_message)
+            .with(
+              hash_including(
+                message_attributes: hash_including(
+                  "request_id" => { data_type: "String", string_value: request_id }
+                )
+              )
+            )
+        end
+      end
+
+      context "with a specified tree_path in Kiev store" do
+        let(:source_tree_path) { "DAaaAAaAAAAAaaAAaaA" }
+
+        before do
+          Kiev::RequestStore.store[:tree_path] = source_tree_path
+          send_message
+        end
+
+        it "appends B to tree_path" do
+          expect(sqs)
+            .to have_received(:send_message)
+            .with(
+              hash_including(
+                message_attributes: hash_including(
+                  "tree_path" => { data_type: "String", string_value: (source_tree_path + "B") }
+                )
+              )
+            )
+        end
+      end
+    end
+
+    describe "server middleware" do
+      it { expect(Shoryuken.server_middleware).to exist(Kiev::Shoryuken::RequestStore) }
+
+      context "when receiving a message" do
+        # include LogHelper
+        # TODO: port/generalize it
+
+        let(:manager)   { double(Shoryuken::Manager) }
+        let(:sqs_queue) { double(Shoryuken::Queue, visibility_timeout: 30) }
+        let(:queue)     { "default" }
+        let(:processor) { Shoryuken::Processor.new(queue, sqs_msg) }
+
+        let(:sqs_msg) do
+          double(
+            Shoryuken::Message,
+            queue_url: queue,
+            body: "{\"a\": 42, \"b\": null}",
+            message_attributes: {},
+            message_id: SecureRandom.uuid,
+            receipt_handle: SecureRandom.uuid
+          )
+        end
+
+        before do
+          class TestWorker
+            include Shoryuken::Worker
+            shoryuken_options queue: "default", body_parser: :json
+
+            def perform(sqs_msg, body)
+              # binding.pry
+              # nil
+            end
+          end
+
+          allow(Shoryuken::Client).to receive(:queues).with(queue).and_return(sqs_queue)
+        end
+
+        context "without message attributes" do
+          it { expect { processor.process }.to_not raise_error }
```

Comment:
Did you see this approach for middleware tests? https://github.com/phstc/shoryuken/blob/master/spec/shoryuken/middleware/server/timing_spec.rb#L45
It looks a bit simpler

Reply:
It does indeed. I'll look into it.

Reply:
Checked this out.
I actually have a stale `let(:manager)` here. I cleaned up some mocks I didn't use, but that has to be the last one.
So aside from this code calling the middleware manually, the only difference is that I use `Sidekiq::Processor` that uses a globally defined middleware stack that's populated by `Kiev::Shoryuken.enable`.
So no, I can't use that approach here.
In middleware unit tests -- yes, sure.